### PR TITLE
STYLE: Use macros in `itk::SLICImageFilter` test

### PR DESCRIPTION
**Pull request by jhlegarreta**
- STYLE: Prefer using macros for `itk::SLICImageFilter` basic methods
- STYLE: Use standard exit return code in `itk::SLICImagetFilter` test
- STYLE: Conform to ITK SWG style guidelines in test ending message

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

File: Modules/Segmentation/SuperPixel/test/itkSLICImageFilterTest.cxx [hunk 1]
```diff
@@ -140,5 +148,6 @@ itkSLICImageFilterTest(int argc, char * argv[])
   }
 
 
-  return 0;
+  std::cout << "Test finished." << std::endl;
```
Comment (blowekamp) [inline Modules/Segmentation/SuperPixel/test/itkSLICImageFilterTest.cxx:151]: I don't think this print statement above is useful

**Review by blowekamp — APPROVED** [review]
Thanks for the clean up!

File: Modules/Segmentation/SuperPixel/test/itkSLICImageFilterTest.cxx (hunk 1 above)
Comment (jhlegarreta) [reply Modules/Segmentation/SuperPixel/test/itkSLICImageFilterTest.cxx:151]: This was brought in PR #3724 and maybe several other PRs as well; [memory was failing me](https://github.com/InsightSoftwareConsortium/ITK/pull/3581#discussion_r955884994) but [Dženan made it clear what I believe is one of the benefits of this] https://github.com/InsightSoftwareConsortium/ITK/pull/3581#discussion_r956041243. Eventually, using a macro that involves the message and the return code (cross-referencing issue #1273) could be beneficial, and bring the debate down to a single line across the whole codebase.

Comment (blowekamp) [reply Modules/Segmentation/SuperPixel/test/itkSLICImageFilterTest.cxx:151]: Just add it here if it is important:
https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/TestKernel/include/itkTestDriverAfterTest.inc#L23

**Pull request merged**